File: benchmark_scripts/triangle_count/py/triangle_count.py
import sys
import time
from scipy.io import mmread
import numpy as np
from math import floor

filename = sys.argv[1]
start_reading = time.time()
G = mmread(filename)
start_compute = time.time()
G_square = G @ G
nb_triangles = G_square.multiply(G).sum() / 3.0
fin = time.time()
duration_reading = fin - start_reading
duration_compute = fin - start_compute
print(f"{duration_reading},{duration_compute},{floor(nb_triangles)}")

File: benchmark_scripts/triangle_count/jl/triangle_count.jl
using MatrixMarket
using SparseArrays

start_reading = time_ns()
G = mmread(ARGS[1])
start_compute = time_ns()
G_square = G * G
nb_triangles = sum(G_square .* G) / 3.0
fin = time_ns()
duration_reading = (fin - start_reading) * 1e-9
duration_compute = (fin - start_compute) * 1e-9
println("$(duration_reading),$(duration_compute),$(floor(Int64, nb_triangles))")
